### PR TITLE
Improvement: Support of needs_string_links in needtable.

### PR DESCRIPTION
**Pull request by iSOLveIT**
- [x] Render ``needs_string_links`` enhanced options also as a clickable link in ``needtable`` columns. Fix for #535. 
- [x] Support for assigning multiple values to a needs_string_links option.

### CODE REVIEW

File: tests/test_needtable.py [hunk 1]
```diff
@@ -76,6 +76,32 @@ def test_doc_needtable_options(test_app):
 
     assert column_order in html
 
+    # Checks for needs_string_links in needtable
+    assert "EXAMPLE_STRING_LINKS" in html
+    assert (
+        '<a class="reference external" href="https://github.com/useblocks/sphinxcontrib-needs/issues/404">'
+        "GitHub #404</a>" in html
```
Comment (danwos) [inline tests/test_needtable.py:83]: I miss the check for the second value `303` here.

**Review by danwos — CHANGES_REQUESTED** [review]
Really great work.
Can you add also a changelog.rst entry for this change?

And I think the tests for the new "additional value" needs some minor additions.

File: tests/test_needtable.py (hunk 1 above)
Comment (iSOLveIT) [reply tests/test_needtable.py:83]: > I miss the check for the second value `303` here.

Changes have been effected.

File: sphinx_needs/utils.py [hunk 2]
```diff
@@ -161,6 +186,29 @@ def row_col_maker(
                 else:
                     ref_col.append(text_col)
                     para_col += ref_col
+            elif matching_link_confs:
+                try:
+                    link_name = None
```
Comment (PhilipPartsch) [inline sphinx_needs/utils.py:191]: Hi @iSOLveIT,
Really cool Idea and implementation.
After a first look, the code here and the code in layout.py codeline 482 mostly look the same.
Is it possible to define a function once and reuse ist in both places?

Comment (iSOLveIT) [reply sphinx_needs/utils.py:191]: Yeah I realised that but I will have to check that out properly. Anyway, thank you for the suggestion.

Comment (iSOLveIT) [reply sphinx_needs/utils.py]: @PhilipPartsch 
I have implemented the suggestions in recent push changes.
Thanks again.

**Review by danwos — APPROVED** [review]
Thanks for the PR and the good work.
Can't wait to see the results in some more complex projects and their tables :)

**Pull request merged**